pkg/ffmpeg/codec_hardware.go

```diff
@@ -337,6 +362,14 @@ func (f *FFMpeg) hwApplyScaleTemplate(sargs string, codec VideoCodec, match []in
 		}
 	case VideoCodecM264:
 		template = "scale_vt=$value"
+	case VideoCodecRK264:
+		if !fullhw {
+			return VideoFilter(sargs)
+		}
+		// Rockchip fallback chain for maximum compatibility:
+		// RGA scale → system memory → upload → rkmpp encoder.
+		// This avoids hwmap(rkrga→rkmpp) failures (-38/-12) seen on some builds.
+		template = "scale_rkrga=$value:format=nv12,hwdownload,format=nv12,hwupload"
```

Comment:
"seen on some builds".
So this bug was fixed? If so you should check which version of ffmpeg this was fixed on.
So we can use the more optimal scale_rkrga